pkg/targets/adapter/awssqstarget/adapter.go

```diff
@@ -107,10 +110,22 @@ func (a *adapter) dispatch(event cloudevents.Event) (*cloudevents.Event, cloudev
 	// The SendMessageInput only accepts a URL for publishing messages. This can be extracted from the ARN
 	url := "https://" + a.awsArn.Service + "." + a.awsArn.Region + ".amazonaws.com/" + a.awsArn.AccountID + "/" + a.awsArn.Resource
 
-	result, err := a.sqsClient.SendMessage(&sqs.SendMessageInput{
-		MessageBody: aws.String(string(msg)),
-		QueueUrl:    &url,
-	})
+	var err error
+	var result *sqs.SendMessageOutput
+	if strings.HasSuffix(url, ".fifo") {
+		ceID := event.ID()
+		result, err = a.sqsClient.SendMessage(&sqs.SendMessageInput{
+			MessageBody:            aws.String(string(msg)),
+			QueueUrl:               &url,
+			MessageGroupId:         &a.messageGroupID,
+			MessageDeduplicationId: &ceID,
```

Comment:
At CoudEvents the combination of [ID and Source attributes is unique](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#required-attributes).